### PR TITLE
Add mca_examples documents for users

### PR DESCRIPTION
**Pull request by luckygalaxy666**
Add `mca_examples.cpp` to demonstrate the use of some mca functions for users
and fix some mistakes in `matrix_construction.cpp`.

See #120.

### CODE REVIEW

File: doc/en/user/mca.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ All the methods are defined in `mca.h`.
 | <nobr>`void powNumber(Matrix<T> &a, const Number &number)`</nobr>                          | `a`'s elements will be the original to the `number`-th power. |
 | <nobr>`void powNumber(const Matrix<T> &a, const Number &number, Matrix<O> &output)`</nobr> | `output`'s elements will be the `a`'s elements to the `nubmer`-th power. |
 
-This part has not been finished yet: add some examples.
+[The examples of mca.](../../../example/mca_examples.cpp)
```
Comment (Kaiser-Yang) [inline doc/en/user/mca.md:43]: use `mca` rather than mca.

File: example/mca_examples.cpp [hunk 2]
```diff
@@ -0,0 +1,181 @@
+
```
Comment (Kaiser-Yang) [inline example/mca_examples.cpp:1]: remove this empty line

File: example/mca_examples.cpp [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    cout << "number * a:\n";
+    cout << number * a << "\n";
+
+    cout << "a / number:\n";
+    cout << a / number << "\n";
+
+    cout << "number / a:\n";
+    cout << number / a << "\n";
+
+    Matrix<double> m(Shape(3, 3), 1);
```
Comment (Kaiser-Yang) [inline example/mca_examples.cpp:75]: rename m with tmp

update m1-m11 with m

File: example/mca_examples.cpp [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    cout << "m10 /= number:\n";
+    m10 /= number;
+    cout << m10 << "\n";
+
+    Matrix<double> m11 = m;
+    cout << "number /= m11:\n";
+    number /= m11;
+    cout << m11 << "\n";
+
+    cout << "Matrix M \n";
```
Comment (Kaiser-Yang) [inline example/mca_examples.cpp:134]: update this M

File: example/mca_examples.cpp [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    Matrix<int> m0({{1, 2, 3}, {4, 5, 6}, {7, 8, 9}});
```
Comment (Kaiser-Yang) [inline example/mca_examples.cpp:135]: update m0 with tmp1

File: example/mca_examples.cpp [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+    cout << m0 << "\n";
+
+    Matrix<int> m12 = m0;
```
Comment (Kaiser-Yang) [inline example/mca_examples.cpp:138]: update m12-14 with m1

**Review by Kaiser-Yang — CHANGES_REQUESTED** [review]

**Review by Quentin9922 — APPROVED** [review]

**Review by Kaiser-Yang — APPROVED** [review]

**Pull request merged**